Source/Details/ASPINRemoteImageDownloader.m

```diff
@@ -116,56 +115,67 @@ + (ASPINRemoteImageDownloader *)sharedDownloader NS_RETURNS_RETAINED
++ (PINRemoteImageManager *)PINRemoteImageManagerWithConfiguration:(nullable NSURLSessionConfiguration *)configuration imageCache:(nullable id<PINRemoteImageCaching>)imageCache
 {
-  NSAssert(!(configuration != nil && preconfiguredPINRemoteImageManager != nil), @"Either configuration or preconfigured image manager can be set at a time.");
-  static dispatch_once_t onceToken;
-  dispatch_once(&onceToken, ^{
-
-    if (preconfiguredPINRemoteImageManager) {
-      sharedPINRemoteImageManager = preconfiguredPINRemoteImageManager;
-    } else {
+  PINRemoteImageManager *manager = nil;
+  // Check that Carthage users have linked both PINRemoteImage & PINCache by testing for one file each
+  if (!(NSClassFromString(@"PINRemoteImageManager"))) {
+    NSException *e = [NSException
+                      exceptionWithName:@"FrameworkSetupException"
+                      reason:@"Missing the path to the PINRemoteImage framework."
+                      userInfo:nil];
+    @throw e;
+  }
+  if (!(NSClassFromString(@"PINCache"))) {
+    NSException *e = [NSException
+                      exceptionWithName:@"FrameworkSetupException"
+                      reason:@"Missing the path to the PINCache framework."
+                      userInfo:nil];
+    @throw e;
+  }
 #if PIN_ANIMATED_AVAILABLE
-      // Check that Carthage users have linked both PINRemoteImage & PINCache by testing for one file each
-      if (!(NSClassFromString(@"PINRemoteImageManager"))) {
-        NSException *e = [NSException
-                          exceptionWithName:@"FrameworkSetupException"
-                          reason:@"Missing the path to the PINRemoteImage framework."
-                          userInfo:nil];
-        @throw e;
-      }
-      if (!(NSClassFromString(@"PINCache"))) {
-        NSException *e = [NSException
-                          exceptionWithName:@"FrameworkSetupException"
-                          reason:@"Missing the path to the PINCache framework."
-                          userInfo:nil];
-        @throw e;
-      }
-      sharedPINRemoteImageManager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration
-                                                                alternativeRepresentationProvider:[self sharedDownloader]];
+    sharedPINRemoteImageManager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration
+                                                                   alternativeRepresentationProvider:[self sharedDownloader]
+                                                                   imageCache:imageCache];
 #else
-      sharedPINRemoteImageManager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration];
+    manager = [[ASPINRemoteImageManager alloc] initWithSessionConfiguration:configuration
```

Comment:
Why did this change to manager instead of `sharedPINRemoteImageManager`?

Reply:
because it's not shared - I funneled all the set-shared logic into as few places (two, unfortunately couldn't get it to one without changing the API of the getter) as possible to reduce complexity. This method just configures a manager and returns it to you.

Reply:
Cool than we should change the other `sharedPINRemoteImageManager` to the `manager` variable too :)